### ISSUE TITLE
(t) request.DATA NotImplementedError

### ISSUE THREAD

**Issue by phillxnet** [rockstor/rockstor-core#2379]
(t) = testing branch. Post our Django and immediate dependencies update in 'Update Django within current Python 2 constrains #2254 ' #2365 We have a deprecated and then removed djangorestframework syntax:

Houston, we've had a problem.
(Failed to config S.M.A.R.T service due to system error.). Lower level exception: (`request.DATA` has been deprecated in favor of `request.data` since version 3.0, and has been fully removed as of version 3.2.). 

```
            Traceback (most recent call last):
  File "/opt/rockstor-dev/src/rockstor/rest_framework_custom/generic_view.py", line 41, in _handle_exception
    yield
  File "/opt/rockstor-dev/src/rockstor/smart_manager/views/smartd_service.py", line 49, in post
    config = request.DATA.get("config", {})
  File "/opt/rockstor-dev/eggs/djangorestframework-3.9.3-py2.7.egg/rest_framework/request.py", line 419, in DATA
    '`request.DATA` has been deprecated in favor of `request.data` '
NotImplementedError: `request.DATA` has been deprecated in favor of `request.data` since version 3.0, and has been fully removed as of version 3.2.
```

## Reproducer
Post resolution of issue #2370 (pr #2380) attempting to configure the S.M.A.R.T service as per:
https://rockstor.com/docs/howtos/smart.html#configure-monitoring
resulting in above Web-UI error.

This same move from request.DATA to request.data in djangorestframework will likely affect a myriad of other configurations.

**Comment by FroggyFlox** [comment]
That's an interesting one! I actually have never seen the `request.DATA` syntax but have only seen `request.data` in what I've looked at. It seems actually that `smartd_service.py` is the only occurrence of this upper case syntax, though, no?
```
$ grep -r "request.DATA" src/rockstor
src/rockstor/smart_manager/views/smartd_service.py:                config = request.DATA.get("config", {})
```

**Comment by phillxnet** [comment]
@FroggyFlox Thanks for taking a look.
Re:

> It seems actually that `smartd_service.py` is the only occurrence of this upper case syntax, though, no?

OK, great. I was embeded in another pr prep so didn't want to get too sidetracked so just created the issue and moved one. Great there is only one. Should be an easy fix then. Nice.

**Comment by phillxnet** [comment]
I'll just prepare a pr to address this so we can tidy it away.

**Comment by phillxnet** [comment]
Closing as:
Fixed by #2381
